### PR TITLE
release: support release archive as Bazel repository

### PR DESCRIPTION
**Pull request by schroederc**
Convert Bazel extraction Docker image to use the Kythe release archive as its embedded tooling repository.

### CODE REVIEW

File: kythe/extractors/bazel/Dockerfile [hunk 1]
```diff
@@ -21,24 +21,19 @@ RUN apt-get update && \
...
-ADD external/io_kythe_lang_proto/kythe/go/extractors/proto/extract_proto_kzip /kythe/extractors/bazel_proto_extractor
-ADD kythe/cxx/extractor/cxx_extractor_bazel /kythe/extractors/bazel_cxx_extractor
-ADD kythe/go/extractors/cmd/bazel/bazel_go_extractor /kythe/extractors/bazel_go_extractor
-ADD kythe/go/extractors/cmd/bazel/extract_kzip /kythe/extractors/bazel_extract_kzip
-ADD kythe/java/com/google/devtools/kythe/extractors/java/bazel/java_extractor_deploy.jar /kythe/extractors/bazel_java_extractor.jar
-ADD kythe/java/com/google/devtools/kythe/extractors/jvm/bazel/bazel_jvm_extractor_deploy.jar /kythe/extractors/bazel_jvm_extractor.jar
+# Extract the Kythe release archive to /kythe
+COPY kythe/release/kythe-v*.tar.gz /tmp/
+RUN tar xzf /tmp/kythe-v*.tar.gz && \
+    mv kythe-v*/ /kythe && \
```
Comment (jaysachs) [inline kythe/extractors/bazel/Dockerfile]: Instead of hardcoding /kythe, let's use an 

   ARG _KYTHE_HOME=/kythe
and
    mv kythe-v*/ ${_KYTHE_HOME} && \

? (Or we can do a separate PR for that.)

Comment (schroederc) [reply kythe/extractors/bazel/Dockerfile]: Why does the location of the repository matter?  It's an internal implementation detail of the Docker container.

Comment (jaysachs) [reply kythe/extractors/bazel/Dockerfile]: Because my thought was that we also include the maven/gradle java artifacts here, and those locations are not self-contained. Clearly this should be discussed outside of this PR.
